### PR TITLE
Completed fetch-in-react-practice exercise. -Lindsey Laveirge

### PR DESCRIPTION
**Pull request by llaveirge**
Functionality: 
fetch-react-practice -vid-1

Database after actions in .gif:
image

### CODE REVIEW

File: fetch-in-react-practice/client/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(newCompletedStatus)
+    })
+      .then(response => response.json())
+      .then(newStatus => {
+        const oldList = this.state.todos;
+        const updatedStatusList = Object.assign(oldList, newStatus);
+        this.setState({ todos: updatedStatusList });
+        this.componentDidMount();
```
Comment (scbowler) [inline fetch-in-react-practice/client/components/app.jsx]: We never want to directly call one of the lifecycle methods. If state is updated properly you wont need this. You can use the `map` method to build a new todos array from the array in state and replace the correct todo item.

```JAVASCRIPT
.then(updated => {
  const allTodos = this.state.todos.map(original => {
    return original.todoId === updated.todoId
      ? updated
      : original;
  });
  this.setState({ todos: allTodos });
});
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just some updates needed for handling the response from updating the todo item. Let me know if you have any questions.

File: fetch-in-react-practice/client/components/app.jsx (hunk 1 above)
Comment (llaveirge) [reply fetch-in-react-practice/client/components/app.jsx]: Sounds good, thanks!

**Comment by llaveirge** [comment]
Still works! 
fetch-react-practice -vid-2

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**